test/with_dummyserver/test_connectionpool.py

```diff
@@ -372,7 +392,7 @@ def test_bad_connect(self):
             pool.request('GET', '/', retries=5)
             self.fail("should raise timeout exception here")
         except MaxRetryError as e:
-            self.assertTrue(isinstance(e.reason, ProtocolError), e.reason)
+            self.assertEqual(type(e.reason), NewConnectionError)
```

Comment:
Of course, pytest can run unittest tests just fine, so you can switch to pytest as your runner and just slowly migrate things over to use regular ``assert`` statements if you want. You don't need to do a big bang up "ok we're on pytest now" thing.